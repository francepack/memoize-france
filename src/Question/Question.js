import React, { Component } from 'react';


export default class Question extends Component {
  constructor(props) {
    super(props);
    this.state = {
      particularQuestion: {}
    }
  }

  render() {
    return(
      <div>
      {this.props.displayQuestion &&
        <div className="question-overlay">
          <article className="question-content">
            <section className="question">
              {this.props.selectedQuestion.question}
            </section>
            <section className="answers">
              {this.props.selectedQuestion.options.map((answerOption, i) => (
                <div className="answer">{answerOption}</div>
              ))}
            </section> 
            <button onClick={this.props.toggleQuestion}>answer</button> 
          </article>
        </div>  
      }  
      </div>    
    )
  }
}