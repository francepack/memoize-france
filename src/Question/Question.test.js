import React from 'react';
import Question from './Question';
import { shallow } from 'enzyme';

import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const evaluateQuestion = jest.fn();

describe('Question', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Question 
        evaluateQuestion={evaluateQuestion}
      />
    )
    wrapper.setProps({ displayQuestion: true, selectedQuestion: {id:'1', category:'People', question:'Who is it?', options:['me', 'you', 'neither', 'both'], answer:'you', link:'http://plcaeholder.com'} });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke evaluateQuestion when an answer is clicked', () => {
    wrapper.find('.answer').at(0).simulate('click', { preventDefault: () => {}});
    expect(evaluateQuestion).toBeCalled(); 
  });

});  