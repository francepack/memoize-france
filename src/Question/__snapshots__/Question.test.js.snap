// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match snapshot 1`] = `
<div>
  <div
    className="question-overlay"
  >
    <article
      className="question-content"
    >
      <section
        className="question"
      >
        Who is it?
      </section>
      <section
        className="answers"
      >
        <div
          className="answer"
          onClick={[MockFunction]}
        >
          me
        </div>
        <div
          className="answer"
          onClick={[MockFunction]}
        >
          you
        </div>
        <div
          className="answer"
          onClick={[MockFunction]}
        >
          neither
        </div>
        <div
          className="answer"
          onClick={[MockFunction]}
        >
          both
        </div>
      </section>
    </article>
  </div>
</div>
`;
